include/boost/di.hpp

```diff
@@ -1712,9 +1712,9 @@ using pool_t = pool<aux::type_list<TArgs...>>;
 template <class... TArgs>
 struct pool<aux::type_list<TArgs...>> : TArgs... {
   template <class... Ts>
-  explicit pool(Ts... args) noexcept : Ts(static_cast<Ts&&>(args))... {}
+  explicit pool(Ts... args) noexcept : TArgs(static_cast<TArgs&&>(args))... {}
```

Comment:
It looks like a magic to me. @krzysztof-jusiak can you explain why does it work?